@Yiyun333 Sorry to say, but the new example is not very usefull.

The original example meant: Limit the string to the first 20 characters (or less if the string is shorter then 20 characters). So a very long string like 'thisisastringwith29characters' would result in: 'thisisastringwith29c'. For a very long string you could simply use substring($verylongstringwithmorethan20characters, 0, 20). For a string shorter then 20 characters (e.g. 5 characters) this would fail: substring($veryshortstringwithlessthan20characters, 0, 20) -> 

```
Caused by: com.mendix.modules.microflowengine.MicroflowException: Failed to evaluate expression, error occurred on line 1, character 1
substring($veryshortstringwithlessthan20characters, 0, 20)
^
	at Test.ACT_substring (SendTextMessage : '{1}')

Advanced stacktrace:
	at com.mendix.languages.mxexpressions.MxExpressionImpl.evaluate(MxExpressionImpl.scala:32)

Caused by: com.mendix.languages.expressions.ExpressionException: begin 0, end 20, length 5
	at com.mendix.languages.expressions.Expr.evaluate(Expr.scala:16)

Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 20, length 5

```

Therefore the example with the min() function was added to the documentation, so this situation will work always:

for a short string (e.g. 5 characters):
substring($veryshortstringwithlessthan20characters, 0, min(length($veryshortstringwithlessthan20characters), 20))
->
substring($veryshortstringwithlessthan20characters, 0, 5)

and for a long string (e.g. more then 20 characters):
substring($verylongstringwithmorethan20characters, 0, min(length($verylongstringwithmorethan20characters), 20))
->
substring($verylongstringwithmorethan20characters, 0, 20)

You'r new example now always returns the complete string, and no longer just a part of the string as expected.

--- a/content/en/docs/refguide/modeling/application-logic/expressions/string-function-calls.md
+++ b/content/en/docs/refguide/modeling/application-logic/expressions/string-function-calls.md
@@ -128,7 +128,7 @@ The output is the following:
 'mystring'
 ```
 
-If you use a third parameter that specifies the desired length of the output:
+If you use a third parameter to specify the desired length of the output:
 
 ```java {linenos=false}
 substring('thisismystring', 6, 2)
@@ -140,10 +140,10 @@ The output is the following:
 'my'
 ```
 
-You can also specify the value of the third parameter using the `length` function:
+To prevent the value of the third parameter from getting out of range, you can set a limit to the third parameter, for instance, using the `min` and `length` functions:
 
 ```java {linenos=false}
-substring('thisismystring', 0, length('thisismystring'))
+substring('thisismystring', 0, min(length('thisismystring'), 20))
 ```
 
 ## 5 find

--- a/content/en/docs/refguide7/desktop-modeler/application-logic/common-elements/expressions/string-function-calls.md
+++ b/content/en/docs/refguide7/desktop-modeler/application-logic/common-elements/expressions/string-function-calls.md
@@ -122,7 +122,7 @@ returns:
 'my'
 ```
 
-If this third parameter is too big, the function will throw an error saying it is out of range. You can specify the value of the third parameter using the `length` function:
+If this third parameter is too big, the function will throw an error saying it is out of range. To prevent the value of the third parameter from getting out of range, you can set a limit to the third parameter, for instance, using the `min` and `length` functions:
 
 ```java {linenos=false}
 substring('thisismystring', 0, min(length('thisismystring'), 20))

--- a/content/en/docs/refguide7/desktop-modeler/application-logic/common-elements/expressions/string-function-calls.md
+++ b/content/en/docs/refguide7/desktop-modeler/application-logic/common-elements/expressions/string-function-calls.md
@@ -124,8 +124,8 @@ returns:
 
 If this third parameter is too big, the function will throw an error saying it is out of range. You can specify the value of the third parameter using the `length` function:
 
-```java
-substring('thisismystring', 0, length('thisismystring'))
+```java {linenos=false}
+substring('thisismystring', 0, min(length('thisismystring'), 20))
 ```
 
 ## 6 find

--- a/content/en/docs/refguide8/modeling/application-logic/expressions/string-function-calls.md
+++ b/content/en/docs/refguide8/modeling/application-logic/expressions/string-function-calls.md
@@ -159,7 +159,7 @@ The output is the following:
 'mystring'
 ```
 
-If you use a third parameter that specifies the desired length of the output:
+If you use a third parameter to specify the desired length of the output:
 
 ```java {linenos=false}
 substring('thisismystring', 6, 2)
@@ -171,10 +171,10 @@ The output is the following:
 'my'
 ```
 
-You can also specify the value of the third parameter using the `length` function:
+To prevent the value of the third parameter from getting out of range, you can set a limit to the third parameter, for instance, using the `min` and `length` functions:
 
 ```java {linenos=false}
-substring('thisismystring', 0, length('thisismystring'))
+substring('thisismystring', 0, min(length('thisismystring'), 20))
 ```
 
 ## 6 find

--- a/content/en/docs/refguide9/modeling/application-logic/expressions/string-function-calls.md
+++ b/content/en/docs/refguide9/modeling/application-logic/expressions/string-function-calls.md
@@ -156,7 +156,7 @@ The output is the following:
 'mystring'
 ```
 
-If you use a third parameter that specifies the desired length of the output:
+If you use a third parameter to specify the desired length of the output:
 
 ```java {linenos=false}
 substring('thisismystring', 6, 2)
@@ -168,10 +168,10 @@ The output is the following:
 'my'
 ```
 
-You can also specify the value of the third parameter using the `length` function:
+To prevent the value of the third parameter from getting out of range, you can set a limit to the third parameter, for instance, using the `min` and `length` functions:
 
 ```java {linenos=false}
-substring('thisismystring', 0, length('thisismystring'))
+substring('thisismystring', 0, min(length('thisismystring'), 20))
 ```
 
 ## 6 find

